src/resolve/adapters/baseSourceAdapter.ts

```diff
@@ -116,6 +116,7 @@ export abstract class BaseSourceAdapter implements SourceAdapter {
     }
   }
 
+  // eslint-disable-next-line class-methods-use-this
```

Comment:
should we consider just setting this to false in the eslint config?

Reply:
There needs to be exceptions for publicly exposed APIs that we don't want to break, but I'd like to keep this pattern from propagating further.